File: Foundation/Arithmetic/sort/insertsort.lua
function insertsort(lists)
    local count = #lists
    for i=2, count do
        key = lists[i]
        j = i - 1
        while j > 0 do
            if lists[j] > key then
                lists[j + 1] = lists[j]
                lists[j] = key
            end
            j = j - 1
        end
    end
    return lists
end

for i,v in ipairs(insertsort{111, 9, 3, 8, 2, 4, 5}) do
    print(v)
end

File: Foundation/Arithmetic/sort/insertsort.php
<?php
$arr = [10, 2, 5, 8, 4, 3, 7, 6, 9];

function insertsort($arr)
{
    for ($i = 1; $i < count($arr); $i++) { 
        if ($arr[$i] < $arr[$i - 1]) {
            $value = $arr[$i];
            for ($j = $i - 1; $j >= 0 && $arr[$j] > $value; $j--) { 
                $arr[$j + 1] = $arr[$j];
            }
            $arr[$j + 1] = $value;
        }
    }
    return $arr;
}

print_r(insertsort($arr));